component-library/src/components/LayerListControls/useLegend.js

```diff
@@ -0,0 +1,38 @@
+/** @typedef {import('./useSelected').SelectedIds} */
+/** @typedef {{ value: String }} ActiveLegend */
+
+import { ref, watch } from '@vue/composition-api'
+import last from '../../utils/last'
+
+export default function useLegend(selectedIds) {
+
+  /** @type {ActiveLegend} */
+  const activeLegend = ref('')
+
+  /**
+   * Set the active legend to a new value, or clear it when the same value is 
+   * provided
+   * @param {String} id 
+   */
+  function setActiveLegend(id) {
+    activeLegend.value = activeLegend.value === id
+      ? ''
+      : id
+  }
+
+  /**
+   * Handler for when `selectedIds` change.
+   * @param {SelectedIds} newSelectedIds
+   */
+  function handleSelectedIdsChange(newSelectedIds) {
+    const lastSelectedId = last(newSelectedIds)
```

Comment:
This should only happen when the activeLegend is not in the selection anymore. Otherwise the activeLegend should just stay the same.